drivers/bluetooth/hci/Kconfig.infineon

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+# an affiliate of Cypress Semiconductor Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Change size of command lengths. It for vendor commands related to
+# firmware downloading.
+config BT_BUF_CMD_TX_SIZE
+	default 255
+
+choice  CYW43XXX_PART
+	prompt "Select CYW43XXX part"
+	depends on BT_CYW43XXX
+
+config CYW4343W
+	bool "CYW4343W"
+	help
+	  Enable Infineon CYW4343W BLE connectivity,
+	  More information about CYW4343W device you can find on
+	  https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/cyw4343w/
+
+config CYW4373
+	bool "CYW4373"
+	help
+	  Enable Infineon CYW4373 BLE connectivity,
+	  More information about CYW4373 device you can find on
+	  https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/cyw4373/
+
+config CYW43012
+	bool "CYW43012"
+	help
+	  Enable Infineon CYW43012 BLE connectivity,
+	  More information about CYW43012 device you can find on
+	  https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/cyw43012/
+
+config CYW43438
+	bool "CYW43438"
+	help
+	  Enable Infineon CYW43438 BLE connectivity,
+	  More information about CYW43438 device you can find on
+	  https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/cyw43438/
+
+config CYW43439
+	bool "CYW43439"
+	help
+	  Enable Infineon CYW43439 BLE connectivity,
+	  More information about CYW43439 device you can find on
+	  https://www.infineon.com/cms/en/product/wireless-connectivity/airoc-wi-fi-plus-bluetooth-combos/cyw43439/
+
+
+choice CYW43012_MODULE
+	prompt "Select CYW43012 module"
+	depends on CYW43012
+	depends on BT_CYW43XXX
+
+config CYW43012_MURATA_1LV
+	bool "MURATA-1LV"
+endchoice
+
+choice CYW4343W_MODULE
+	prompt "Select CYW4343W module"
+	depends on CYW4343W
+	depends on BT_CYW43XXX
+
+config CYW4343W_MURATA_1DX
+	bool "MURATA-1DX"
+endchoice
+
+choice CYW4373_MODULE
+	prompt "Select CYW4373 module"
+	depends on CYW4373
+	depends on BT_CYW43XXX
+
+config CYW4373_STERLING_LWB5PLUS
+	bool "STERLING-LWB5plus"
+endchoice
+
+choice CYW43439_MODULE
+	prompt "Select CYW43439 module"
+	depends on CYW43439
+	depends on BT_CYW43XXX
+
+config CYW43439_MURATA_1YN
+	bool "MURATA_1YN"
+endchoice
+
+config CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB
```

Comment:
Where and how is this option used? I don't see any references to it in this PR.

Reply:
i will push one more commit with zephyr\modules\hal_infineon\btstack-integration\CMakeLists.txt  after https://github.com/zephyrproject-rtos/zephyr/pull/44211.

Zephyr\modules\hal_infineon\btstack-integration\CMakeLists.txt will use this.
So user in prj.conf should provide path to HCD blob file:

it can be relative path from Application folder:
```
 CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB="BCM4343A1_001.002.009.0153.0000_Generic.hcd"
```
or absolute path:
```
CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB="c:/cyw43xxx_fw/BCM4343A1_001.002.009.0153.0000_Generic.hcd"
```

zephyr\modules\hal_infineon\btstack-integration\CMakeLists.txt will check CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB and generate c-array. See fragment of cmake below, full version i attached in post https://github.com/zephyrproject-rtos/zephyr/pull/55014#discussion_r1113610804:

```cmake
....
....
# use user provided FIRMWARE HCD file (path must be defined in CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB)
if(CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB)
  # Allowed to pass absolute path to HCD blob file, or relative path from Application folder. 
  if(IS_ABSOLUTE ${CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB}) 
    set(blob_hcd_file  ${CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB})
  else()
    set(blob_hcd_file  ${APPLICATION_SOURCE_DIR}/${CONFIG_CYW43XX_CUSTOM_FIRMWARE_HCD_BLOB})
  endif()
endif()

# generate Bluetooth include blob from HCD binary
if(EXISTS ${blob_hcd_file})
  message(INFO " generate include of blob Bluetooth file: ${blob_hcd_file}")

  generate_inc_file_for_target(app ${blob_hcd_file} ${blob_gen_inc_file})
  zephyr_library_sources(${CMAKE_CURRENT_SOURCE_DIR}/w_bt_firmware_controller.c)
endif()
```

Reply:
added commits 2424db13275b547fb74fe8ebca1b3812d1ac982e with Cmake where this option is used.